quickfixj-core/src/main/java/quickfix/SessionSettings.java

```diff
@@ -414,6 +439,12 @@ private void loadFromFile(String filename) throws ConfigError {
         }
     }
 
+    private void loadFromList(List<String> listValues) throws ConfigError {
+        byte[] bytes = listValues.stream().collect(Collectors.joining("\n")).getBytes();
```

Comment:
The line separator needs to be changed so that it can work with Unix and Windows line endings. The line separator of course depends on where the used config file originates from.
But since `SessionSettings` itself can work with both types of line separators (see how `NEWLINE` constant is used) we should not restrict it in this method.

Reply:
@chrjohn luckly you noticed this problem with the line separators, I hadn't thought of that.

I changed from `\n` to `System.lineSeparator()`. I believe it will work, but I must test it in Windows first to make sure it will work. I'll let you know here after I get this test with Windows done. In Linux it worked just fine.

But, if the `System.lineSeparator()` wasn't the right solution either, just say and I'll think in other thing.

I didn't want to polute the git log, so I did a rebase and squashed my commits. Please tell me what is best to do after a review? Should I add a new commit or should I rebase it and squash my commits? If you prefer, I can cancel this PR and open a new one.

Thanks in advance.

Reply:
The commits are fine with me, thanks.

Maybe you can simply convert your input file with a tool like `unix2dos` or similar.

Reply:
Thanks for the tip about `unix2dos`. I'll try use it later.

I had already set up one Windows environment and did the tests with this new constructor and everything worked fine.

So, by my tests, the constructor worked fine in Linux and Windows.

For my point of view, it's all ok now.

If you want another change, please say!

Thanks.

Reply:
Turned out I misunderstood the part about the `Collectors.joining(\n)`. At first I thought this was the delimiter that is used in the settings input file, but it is of course the delimiter between the values of the passed list. However, my comment about using `System.lineSeparator()` is still valid. So in the end everything is fine. :)